omero/developers/Java.txt

```diff
@@ -662,11 +677,33 @@ to an image.
     rectangle.setT(0);
     data.addShapeData(rectangle);
 
+    //Add a mask
+    PixelsData pixels = image.getDefaultPixels();
+    long pixelsId = pixels.getId();
+    RawPixelsStorePrx store = gateway.getPixelsStore(ctx);
```

Comment:
sorry I missed that before,
you should add a try/finally block otherwise the store might never be closed
i.e.
``
try {
store.setPixelsId(pixelsId, false);
 byte[] mask = store.getStack(0, 0);
MaskData maskData = new MaskData(10, 10, 100.0, 100.0, mask);
maskData.setZ(0);
maskData.setT(0);
data.addShapeData(maskData);
} finally {
store.close();
}

Add do the same in other location where the store is initialised and closed